lib/perl/Genome/Model/ReferenceAlignment/Command/BsmapMethCalcConversionRate.pm

```diff
@@ -0,0 +1,100 @@
+package Genome::Model::ReferenceAlignment::Command::BsmapMethCalcConversionRate;
+
+use strict;
+use warnings;
+use Genome;
+use Genome::Model::Tools::Bsmap::MethCalcConversionRate;
+
+class Genome::Model::ReferenceAlignment::Command::BsmapMethCalcConversionRate {
+    is => 'Command::V2',
+    has => [
+        model => {
+			is => 'Genome::Model::ReferenceAlignment',
+            doc => 'Use genome model ID to calculate methylation conversion',
+			shell_args_position => 1,
+        },
+        output_file => {
+            is => 'String',
+            is_optional => 1,
+            doc => 'Output methylation conversion',
+			shell_args_position => 2,
+        },
+    ],
+};
+
+sub help_synopsis {
+  return <<EOS
+    genome model reference-alignment bsmap-meth-calc-conversion-rate --model-id=394d6228a7b5487a9cb0ad0c448b5a44
+EOS
+}
+
+sub help_brief {
+    "calculate methylation conversion using Mitochondrial DNA or spiked lambda"
+}
+
+sub help_detail {
+    "calculate methylation conversion using Mitochondrial DNA or spiked lambda"
+}
+
+sub execute {
+    my $self = shift;
+    my $output_file = $self->output_file;
+
+	my $cfile;
+	if ($output_file) {
+		open($cfile, '>', $output_file) or die;
```

Comment:
We generally prefer to use the subroutines provided in Genome::Sys for things like these. It's a good package to familiarize yourself with as it provides a lot of functionality. Here you can use Genome::Sys->open_file_for_writing. It already does the error handling for you.